### PR TITLE
Bug 1414107: Update typography hierarchy

### PR DESCRIPTION
**Pull request by stephaniehobson**
- add separator variables
- add separator to h2s
- add separator to end of articles
- make hr match separator
- collapse separator lines when appearing next to each other:
  - summaries, blockquotes, top of the page, hrs
- move highlight to h3 and h5
- font size and margin changes to all headings
- remove empty p and div tags from article bodies
- adjust card-grid to appear better as full-width-content
- tweaks to components that use headings in them to accommodate new
  margins and typography on headings

### CODE REVIEW

File: kuma/static/styles/components/content.scss [hunk 1]
```diff
@@ -81,10 +81,57 @@ We cannot put these styles in the base/elements/ files because they don't apply
...
+    article[id=wikiArticle] > h2,
+    article[id=wikiArticle] > div:not([class]) > h2 {
+        @include heading-2-section();
+    }
+
+    /*  remove top spacing if h2 is effectivly first element in article */
+    article[id=wikiArticle] > h2:first-child,
+    article[id=wikiArticle] > div:first-child:empty + h2,
+    article[id=wikiArticle] > p:first-child:empty + h2,
+    .Quick_links + h2 {
```
[the remark this answers is not in the document]
Comment (stephaniehobson) [reply kuma/static/styles/components/content.scss]: No.

The macros that create the quicklinks menus do so by inserting a `<div class="Quick_links">` into the wiki article. When kuma processes it, it takes the contents of the div and moves it to the left column, but leaves the div behind. 😖

The `<div class="Quick_links">` is then the first child of the article.

Often the  `<div class="Quick_links">` itself is wrapped in a `<div>` by CKEditor. Without a class I can't target those instances though 🙁

File: kuma/static/styles/base/elements/typography.scss [hunk 2]
```diff
@@ -14,24 +14,35 @@ h1 {
...
+    @include set-font-size($h5-font-size);
+    @include set-heading-font-family();
+    line-height: 1.2;
+}
+
+h6 {
+    margin: ($grid-spacing / 2) 0;
     @include set-font-size($body-font-size);
     @include set-heading-font-family();
+    line-height: 1.2;
```
[the remark this answers is not in the document]
Comment (stephaniehobson) [reply kuma/static/styles/base/elements/typography.scss]: I like having all the styles for one element in one place and with gzip I don't think there's much file size savings to be gained. You make a good point about them all repeating the same thing, I have abstracted it to a variable.

File: kuma/static/styles/components/content.scss [hunk 3]
```diff
@@ -81,10 +81,57 @@ We cannot put these styles in the base/elements/ files because they don't apply
...
+    }
+
+    h3,
+    h5 {
         @extend %highlight;
     }
 
+    h3,
+    h3 .highlight-span {
+        @include bidi-value(padding, 0 4px 0 $grid-spacing, 0 $grid-spacing 0 4px);
```
Comment (schalkneethling) [inline kuma/static/styles/components/content.scss]: This turns into `padding: 0 4px 0 40px;` - is that intentional or should it be `padding: 0 4px 0 4px;`?

screen shot 2017-11-08 at 14 27 13

Comment (stephaniehobson) [reply kuma/static/styles/components/content.scss]: That's intentional. You can see the indentation on headings on moz org, like here: https://www.mozilla.org/en-US/firefox/

**Pull request merged**